src/pip/_internal/operations/install/wheel.py

```diff
@@ -324,6 +327,110 @@ def make(self, specification, options=None):
         return super(PipScriptMaker, self).make(specification, options)
 
 
+def _get_file_owners(
+        lib_dir,      # type: str
+        top_level,    # type: str
+        ignore_name,  # type: str
+):
+    # type: (...) -> Dict[str, List[str]]
+    """Return dict mapping distributions to files under a top level directory
+
+    Look through lib_dir for distributions that own files under the
+    top_level directory specified. Skip distributions that match the
+    ignore_name. Return a dict mapping filenames to the distributions
+    that own them.
+
+    """
+    file_owners = {}  # type: Dict[str, List[str]]
+    existing_env = pkg_resources.Environment([lib_dir])
+    canonical_name = canonicalize_name(ignore_name)
+    for existing_dist_name in existing_env:
+        for d in existing_env[existing_dist_name]:
+            if canonicalize_name(d.project_name) == canonical_name:
+                continue
+
+            existing_info_dir = os.path.join(
+                lib_dir,
+                '{}-{}.dist-info'.format(d.project_name, d.version),
+            )
+
+            top_level_path = os.path.join(existing_info_dir, 'top_level.txt')
+            if not os.path.exists(top_level_path):
+                continue
+            with open(top_level_path, 'r') as f:
+                existing_top_level = f.read().strip()
+            if existing_top_level != top_level:
+                continue
+
+            record_path = os.path.join(existing_info_dir, 'RECORD')
+            if not os.path.exists(record_path):
+                continue
+            with open(record_path, **csv_io_kwargs('r')) as record_file:
+                for row in csv.reader(record_file):
+                    # Normalize the path before saving the owners,
+                    # since the record always contains forward
+                    # slashes, but when we look for a path later we
+                    # will be using a value with the native path
+                    # separator.
+                    o = file_owners.setdefault(os.path.normpath(row[0]), [])
+                    o.append(str(d))
+
+    return file_owners
+
+
+def _report_file_owner_conflicts(
+        lib_dir,     # type: str
+        name,        # type: str
+        source_dir,  # type: str
+        info_dir,    # type: str
+):
+    # type: (...) -> None
```

Comment:
Same Nitpick here as before.

```suggestion
def _report_file_owner_conflicts(lib_dir, name, source_dir, info_dir):
    # type: (str, str, str, str) -> None
```